app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/viewmodels/IdentifyResultViewModel.kt

```diff
@@ -19,38 +19,48 @@ package com.esri.arcgisruntime.opensourceapps.datacollection.viewmodels
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
 import com.esri.arcgisruntime.data.Feature
 import com.esri.arcgisruntime.layers.FeatureLayer
 import com.esri.arcgisruntime.layers.LayerContent
 import com.esri.arcgisruntime.mapping.GeoElement
-import com.esri.arcgisruntime.mapping.popup.Popup
 import com.esri.arcgisruntime.mapping.view.IdentifyLayerResult
 import com.esri.arcgisruntime.opensourceapps.datacollection.util.Event
 import com.esri.arcgisruntime.opensourceapps.datacollection.util.raiseEvent
 
 /**
  * The view model for IdentifyResultFragment, that is responsible for processing the result of
- * identify layer operation on MapView and displaying the values of the display fields of result
- * Popup in the bottom sheet.
+ * identify layer operation on MapView, highlighting the selected feature and displaying the values
+ * of the display fields of result Popup in the bottom sheet.
  */
-class IdentifyResultViewModel : ViewModel() {
+class IdentifyResultViewModel(val popupViewModel: PopupViewModel) : ViewModel() {
 
-    private val _showIdentifiedPopupAttributeEvent = MutableLiveData<Event<Unit>>()
-    val showIdentifiedPopupAttributeEvent: LiveData<Event<Unit>> = _showIdentifiedPopupAttributeEvent
+    private val _showIdentifyResultFragmentEvent = MutableLiveData<Event<Unit>>()
+    val showIdentifyResultFragmentEvent: LiveData<Event<Unit>> = _showIdentifyResultFragmentEvent
 
     private val _identifyLayerResult = MutableLiveData<IdentifyLayerResult>()
     val identifyLayerResult: LiveData<IdentifyLayerResult> = _identifyLayerResult
 
-    private val _identifiedPopup = MutableLiveData<Popup>()
-    val identifiedPopup: LiveData<Popup> = _identifiedPopup
+    private val _showPopupFragmentEvent = MutableLiveData<Event<Unit>>()
+    val showPopupFragmentEvent: LiveData<Event<Unit>> = _showPopupFragmentEvent
 
-    private val _showPopupAttributeListEvent = MutableLiveData<Event<Unit>>()
-    val showPopupAttributeListEvent: LiveData<Event<Unit>> = _showPopupAttributeListEvent
+    /**
+     * Factory class to help create an instance of [IdentifyResultViewModel] with a [PopupViewModel].
+     */
+    class Factory(private val popupViewModel: PopupViewModel) :
+        ViewModelProvider.Factory {
+        override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+            return IdentifyResultViewModel(popupViewModel) as T
+        }
+    }
 
     /**
-     * Highlights the result popup in the GeoView.
-     * Updates identifiedPopup property to set the popup field values being displayed in
-     * the bottom sheet.
+     * <ul>
+     * <li>Sets the identifyLayerResult instance on the identifyResultViewModel
+     * <li>Sets the first popup returned by identifyLayerResult on PopupViewModel
+     * <li>Raises the event to show the popup in IdentifyResultFragment in the BottomSheet
```

Comment:
I would only mention: "Raises an event to show the identify result"

How this identify result is actually presented, i.e. in a Fragment in a bottom sheet, is not the concern of the view model.

Reply:
updated